SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt

```diff
@@ -108,317 +102,442 @@ class TradeFlow @JvmOverloads constructor(
-                        append(stringResource(id = R.string.trade_flow_text_field_amount_placeholder))
-                        withStyle(style = SpanStyle(
-                            color = colorResource(id = R.color.list_prices_asset_component_code_color))) {
-                            append(" ${valueLabelHintAsset.code}")
-                        }
-                    }
-                    val amountLabel = buildAnnotatedString {
-                        append(stringResource(id = R.string.trade_flow_text_field_amount_placeholder))
-                        append(" ${valueLabelHintAsset.code}")
-                    }
 
-                    // -- DropDown
-                    val expanded = remember { mutableStateOf(false) }
-                    val textFieldSize = remember { mutableStateOf(Size.Zero) }
-                    val selectedTabIndex = remember { mutableStateOf(0) }
-                    val icon = Icons.Filled.ArrowDropDown
-
-                    TabRow(
-                        selectedTabIndex = selectedTabIndex.value,
-                        backgroundColor = Color.Transparent,
-                        indicator = { tabsIndicators ->
-                            Box(
-                                Modifier
-                                    .tabIndicatorOffset(tabsIndicators[selectedTabIndex.value])
-                                    .height(2.dp)
-                                    .border(3.5.dp, colorResource(id = R.color.primary_color))
+                    // -- Currency Input --> DropDown
+                    val expandedCurrencyInput = remember { mutableStateOf(false) }
+                    val currencyInputWidth = remember { mutableStateOf(Size.Zero) }
+
+                    // -- Value Input
+                    val typeOfValueState = remember { mutableStateOf(AssetBankModel.Type.fiat) }
+                    val valueState = remember { mutableStateOf("") }
+                    val valueAsset = remember { mutableStateOf(
+                        if (typeOfValueState.value == AssetBankModel.Type.fiat)
+                            pairAsset else currencyState.value
+                    )}
+
+                    CryptoHeaderTabs(
+                        selectedTabIndex = selectedTabIndex,
+                        tabs = tabs
+                    )
+
+                    CryptoCurrencyInput(
+                        currencyState = currencyState,
+                        expandedCurrencyInput = expandedCurrencyInput,
+                        currencyInputWidth = currencyInputWidth
+                    )
+
+                    CryptoCurrencyDropDown(
+                        currencyState = currencyState,
+                        expandedCurrencyInput = expandedCurrencyInput,
+                        currencyInputWidth = currencyInputWidth,
+                        cryptoList = cryptoList
+                    )
+
+                    if (!expandedCurrencyInput.value) {
```

Comment:
What's meant by `expandedCurrencyInput`?

Reply:
This val is a:

`val expandedCurrencyInput = remember { mutableStateOf(false) }`

It means that handle if the curreny dropdownmenu its open or not

![Captura de Pantalla 2022-06-17 a la(s) 13 33 49](https://user-images.githubusercontent.com/7268597/174357548-b9ff340c-84ce-4dae-9935-e2f774f74564.png)